examples/daymet-hi-annual.json

```diff
@@ -0,0 +1,186 @@
+            "axis": "x",
+            "description": "North American Lambert Conformal Conic, 1KM grid.",
+            "extent": [
+                -622000,
+                -5519250
+            ],
+            "step": 1000,
+            "reference_system": "PROJCRS[\"undefined\",BASEGEOGCRS[\"undefined\",DATUM[\"undefined\",ELLIPSOID[\"undefined\",6378137,298.257223563,LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]]],PRIMEM[\"Greenwich\",0,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8901]]],CONVERSION[\"unknown\",METHOD[\"Lambert Conic Conformal (2SP)\",ID[\"EPSG\",9802]],PARAMETER[\"Latitude of 1st standard parallel\",25,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8823]],PARAMETER[\"Latitude of 2nd standard parallel\",60,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8824]],PARAMETER[\"Latitude of false origin\",42.5,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8821]],PARAMETER[\"Longitude of false origin\",-100,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8822]],PARAMETER[\"Easting at false origin\",0,LENGTHUNIT[\"metre\",1],ID[\"EPSG\",8826]],PARAMETER[\"Northing at false origin\",0,LENGTHUNIT[\"metre\",1],ID[\"EPSG\",8827]]],CS[Cartesian,2],AXIS[\"(E)\",east,ORDER[1],LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]],AXIS[\"(N)\",north,ORDER[2],LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]]]"
+        },
+        "y": {
+            "type": "spatial",
+            "axis": "y",
+            "description": "North American Lambert Conformal Conic, 1KM grid.",
+            "extent": [
+                -5802250,
+                -39000
+            ],
+            "step": 1000,
+            "reference_system": "PROJCRS[\"undefined\",BASEGEOGCRS[\"undefined\",DATUM[\"undefined\",ELLIPSOID[\"undefined\",6378137,298.257223563,LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]]],PRIMEM[\"Greenwich\",0,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8901]]],CONVERSION[\"unknown\",METHOD[\"Lambert Conic Conformal (2SP)\",ID[\"EPSG\",9802]],PARAMETER[\"Latitude of 1st standard parallel\",25,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8823]],PARAMETER[\"Latitude of 2nd standard parallel\",60,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8824]],PARAMETER[\"Latitude of false origin\",42.5,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8821]],PARAMETER[\"Longitude of false origin\",-100,ANGLEUNIT[\"degree\",0.0174532925199433],ID[\"EPSG\",8822]],PARAMETER[\"Easting at false origin\",0,LENGTHUNIT[\"metre\",1],ID[\"EPSG\",8826]],PARAMETER[\"Northing at false origin\",0,LENGTHUNIT[\"metre\",1],ID[\"EPSG\",8827]]],CS[Cartesian,2],AXIS[\"(E)\",east,ORDER[1],LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]],AXIS[\"(N)\",north,ORDER[2],LENGTHUNIT[\"metre\",1,ID[\"EPSG\",9001]]]]"
+        },
+        "nv": {
+            "type": "count?",
+            "description": "size of time_bnds",
+            "values": [
+                0,
+                1
+            ]
+        }
+    },
+    "cube:coordinates": {
```

Comment:
What is cube:coordinates?

Reply:
This is currently invalid, according to the updated schema.

As described in https://xarray.pydata.org/en/stable/user-guide/terminology.html, this is a nnon-dimension-coordinate. https://xarray.pydata.org/en/stable/examples/multidimensional-coords.html has an example, but it's trying to describe an n-dimensional array that's really a coordinate (rather than an actual data value). The example there is something like latitude, which will depend on the `(y, x)` coordinates.

My suggestion: just put it in cube:variables and somehow mark it as a coordinate with a `type` field that's either `data` or `coordinate`. Does that sound resonable?

Reply:
I've found another comment above that describes the issue, I've replied there:

> In terms of this extension, coordinates are "Additional dimensions" with type set to "coordinate" (or so), If I understand it correctly.

Reply:
Yes, now that I look again I think your right... Even in the xarray repr above (https://github.com/stac-extensions/datacube/pull/6#issue-643216316) `lat` is under "Dimensions" rather than "Data variables". My apologies.

One thing, though, do you have thoughts on *allowing* (not requiring) the objects in `cube:coordinates` to include a `dimensions` field? To correctly describe this dataset, that would be required.

Reply:
> One thing, though, do you have thoughts on _allowing_ (not requiring) the objects in `cube:coordinates` to include a `dimensions` field? To correctly describe this dataset, that would be need to be allowed.

I'm confused. I thought coordinates does away and will be dimensions? And is this the same question as discussed in https://github.com/stac-extensions/datacube/pull/6#issuecomment-839843274, last part?

Reply:
Sorry, I meant allowing `cube:dimensions` objects to themselves have `dimensions`. You're correct that in my latest commit `cube:coordinates` is gone.

> I'm actually also confused what "coordinates" are useful for? Isn't it implicitly clear that the spatial dimensions form the coordinates?

Maybe a bit clearer to say that the *values* along the spatial (and time, and additional) dimensions form the coordinates. The coordinates are the actual labels / values along that dimension.

In my [example dataset](https://github.com/stac-extensions/datacube/pull/6#issue-643216316), things like `latitude` and `longitude` aren't actually a dimension of any of the variables (note in the repr that none the variables include them as a dimension; e.g. `prcp` has `(time, y, x)`). The netCDF docs call this an [auxiliary coordinate](https://unidata.github.io/python-training/workshop/CF%20Conventions/netcdf-and-cf-the-basics/#Auxilliary-Coordinates) and xarray calls it a [non-dimension coordinate](https://xarray.pydata.org/en/stable/user-guide/terminology.html#term-Non-dimension-coordinate).

I don't know if @rabernat has thoughts here or can ping an xarray / netCDF developer who can explain this better than I am. I'm still hazy on these concepts.

Reply:
Me neither, our data cubes don't have these concepts, we just have dimensions (and potentially variables derived from these dimensions), but I don't really get the point behint those coordinates.

Reply:
The concept of coordinates come from the [CF Conventions](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#coordinate-types), the gold standard for metadata in weather and climate science.

> The commonest use of coordinate variables is to locate the data in space and time, but coordinates may be provided for any other continuous geophysical quantity (e.g. density, temperature, radiation wavelength, zenith angle of radiance, sea surface wave frequency) or discrete category (see Section 4.5, "Discrete Axis", e.g. area type, model level number, ensemble member number) on which the data variable depends.

A special type of coordinate is a "coordinate variable", which is 1D and has the same name as a dimension. But coordinates can be ND. CF calls the "auxiliary coordinate variables". For example, for a 3D datacube of temperature on a curvilinear grid, the dimensions might be `time, y, x`, and we would typically have coordinates `lat(y, x)` and `lon(y, x)`. Xarray calls this a non-dimension coordinate. Note that this treatment of coordinates is considerably different from the typical geospatial raster scenario, which would instead specify a projection rather than explicit arrays of lon / lat. But this is ubiquitous in weather / climate data.

At a pure data level, coordinates are just variables. But at a metadata level, they are treated differently by analysis libraries. For example, in xarray, if I add two datasets together

```python
ds1 + ds2
```

Their data variables will be added, but their coordinates will not. It would not make sense to add two latitudes.

Pure NetCDF and Zarr (without CF conventions) do not have an explicit way to mark a variable as a coordinate. Xarray follows CF conventions and decodes a variable to the list of coordinates (as opposed to variables) if it meets either of the two criteria:
- it is 1D and has the same name as a dimension (coordinate variable)
- it appears in another variables `coordinates` attribute (auxiliary coordinate variable)

I am not certain that the notion of CF coordinates needs to be in the datacube extension. To answer this, I would want to better understand the relationship between this extension and the CF conventions.